doc: Rename to Nixpkgs reference manual and restate purpose (#272183)
NixOS/nixpkgs: doc/preface.chapter.md

Packages are available for several platforms, and can be used with the Nix
package manager on most GNU/Linux distributions as well as [NixOS](https://nixos.org/nixos).

This document is the user [_reference_](https://nix.dev/contributing/documentation/diataxis#reference) manual for Nixpkgs.

> Its purpose is to contain consise and complete listings of all functions, arguments and attributes with examples and cross-links.

Reviewer: What about behavior?

```suggestion
Its purpose is to contain consise and complete listings of all functions, arguments, attributes and behaviors with examples and cross-links.
```

Reviewer: ```suggestion
Its purpose is to describe the Nixpkgs machinery, in a concise and orderly manner, the entire public interface and all relevant behaviors, with examples and cross links.
```

I don' think we have to list out what kinds of things the public interface may consist of. What's important here is completeness.
I'm including behaviors because we have a bunch of things that aren't purely functional, such as `trace`, `setup.sh`, and all the hooks.

Reviewer: A preface seems particularly prone to being overlooked. Perhaps we can focus attention on the relevant keywords.

```suggestion
Its purpose is to describe the **Nixpkgs machinery**, in a **concise** and orderly manner, the **entire** public interface and all relevant behaviors, with examples and cross links.
```

Reviewer 2: I like the suggestion, except "Nixpkgs machinery".

Author: Agreeing with @fricklerhandwerk's final suggestion, applying!

To discover other kinds of documentation:

> - [nix.dev](https://nix.dev/): Documentation entry-point including tutorials and guides for the Nix package manager and the Nix expression language

Reviewer: ```suggestion
- [nix.dev](https://nix.dev/): Tutorials and guides for learning about and achieving goals with Nix
```

Reviewer 2: Why not build on the upstream tagline: 

```suggestion
- [nix.dev](https://nix.dev/): Tutorials and guides for getting things done with Nix
```

Author: Applied @fricklerhandwerk's final suggestion

Reviewer: Not sure if _learning_ is _getting things done_, but close enough.

> - [nix.dev](https://nix.dev/): Documentation entry-point including tutorials and guides for the Nix package manager and the Nix expression language

Reviewer 3: Link to the actual Nix manual here.

Author: I don't think that makes sense, `nix.dev` is not just for Nix itself. Shouldn't be a concern anymore with the updated description of nix.dev from https://github.com/NixOS/nixpkgs/pull/272183#discussion_r1416460985

> - [search.nixos.org](https://search.nixos.org): Nixpkgs package and NixOS option search

Reviewer: ```suggestion
- [NixOS **Option Search**](https://search.nixos.org/options) and reference documentation
- [Nixpkgs **Package Search**](https://search.nixos.org/packages)
```

> - [NixOS manual](https://nixos.org/manual/nixos/stable/): Reference documentation for the NixOS Linux distribution

Reviewer: ```suggestion
- [**NixOS** manual](https://nixos.org/manual/nixos/stable/): Reference documentation for the NixOS Linux distribution
```
